tests/BiosTests.hs

```diff
@@ -195,15 +206,38 @@ findCradleForModule fp expected' step = do
   step "Finding cradle"
   findCradle a_fp `shouldReturn` expected
 
-testImplicitCradle :: FilePath -> ActionName Void -> (String -> IO ()) -> IO ()
-testImplicitCradle fp' expectedActionName step = do
-  fp <- canonicalizePath fp'
+testImplicitCradle :: FilePath -> ActionName Void -> FilePath -> (String -> IO ()) -> IO ()
+testImplicitCradle fp' expectedActionName expectedCradleRootDir step = do
```

Comment:
Maybe second and third parameter should be in some correlation? E.g. we have a project dir and the filepath to load is relative to the given project dir. Otherwise we repeat ourselves quite often.